satpy/readers/__init__.py

```diff
@@ -590,6 +583,32 @@ def load_readers(filenames=None, reader=None, reader_kwargs=None):
     return reader_instances
 
 
+def _get_readers_files(filenames, reader, idx, remaining_filenames):
+    if isinstance(filenames, dict):
+        return set(filenames[reader[idx]])
+    return remaining_filenames
+
+
+def _get_reader_instance(reader, reader_configs, idx, reader_kwargs):
+    reader_instance = None
+    try:
+        reader_instance = load_reader(
+            reader_configs,
+            **reader_kwargs[None if reader is None else reader[idx]])
+    except (KeyError, IOError) as err:
```

Comment:
Don't we want to keep the YamlError here, in case something else happens in the yaml file?

Reply:
I thought you wanted to get rid of it in the earlier comment in favor of a more specific one.

Reply:
I do want the specific one to detect the import error.
But if we have another kind of error in the yaml, we don't catch it anymore. (but maybe it's a good thing?)

Reply:
Short discussion on slack conclude this is good for now.